src/functions.js

```diff
@@ -0,0 +1,129 @@
+    return;
+  }
+  todoListElement.innerHTML = '';
+  todos.forEach((todo, index) => {
+    todoListElement.innerHTML += `<div class='check-list'><div class="flex-container" id='${index}'>
+    <div class="left">
+    <input data-action="check" class="checkbox" type="checkbox">
+    <div data-action = 'edit' >${todo.value}</div>
+    </div>
+    <div data-action="edit" class="dots"><svg xmlns="http://www.w3.org/2000/svg" data-action='edit' width="16" height="16" fill="currentColor" class="bi bi-pencil-square editsvg" viewBox="0 0 16 16">
+    <path d="M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z"/>
+    <path fill-rule="evenodd" d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z"/>
+  </svg><svg xmlns="http://www.w3.org/2000/svg" data-action='delete'  width="16" height="16" fill="currentColor" class="bi bi-trash3 deletesvg" viewBox="0 0 16 16">
+    <path d="M6.5 1h3a.5.5 0 0 1 .5.5v1H6v-1a.5.5 0 0 1 .5-.5ZM11 2.5v-1A1.5 1.5 0 0 0 9.5 0h-3A1.5 1.5 0 0 0 5 1.5v1H2.506a.58.58 0 0 0-.01 0H1.5a.5.5 0 0 0 0 1h.538l.853 10.66A2 2 0 0 0 4.885 16h6.23a2 2 0 0 0 1.994-1.84l.853-10.66h.538a.5.5 0 0 0 0-1h-.995a.59.59 0 0 0-.01 0H11Zm1.958 1-.846 10.58a1 1 0 0 1-.997.92h-6.23a1 1 0 0 1-.997-.92L3.042 3.5h9.916Zm-7.487 1a.5.5 0 0 1 .528.47l.5 8.5a.5.5 0 0 1-.998.06L5 5.03a.5.5 0 0 1 .47-.53Zm5.058 0a.5.5 0 0 1 .47.53l-.5 8.5a.5.5 0 1 1-.998-.06l.5-8.5a.5.5 0 0 1 .528-.47ZM8 4.5a.5.5 0 0 1 .5.5v8.5a.5.5 0 0 1-1 0V5a.5.5 0 0 1 .5-.5Z"/>
+  </svg><svg xmlns="http://www.w3.org/2000/svg"  width="16" height="16" fill="currentColor" class="bi bi-three-dots-vertical dotssvg" viewBox="0 0 16 16">
+    <path d="M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z"/>
+  </svg></div>
+</div></div>`;
+  });
+}
+// add an event listener to clicks on the to do list
+
+todoListElement.addEventListener('click', (event) => {
+  /* eslint-disable prefer-destructuring */
+  const target = event.target;
+  const parentElement = target.parentNode.parentNode;
+  if (parentElement.className !== 'flex-container') return;
+  const todo = parentElement;
+  const todoId = Number(todo.id);
+
+  // data action
+  /* eslint-disable prefer-destructuring */
+  const action = target.dataset.action;
+  // action === 'check' && checkTodo(todoId);
+
+  /* eslint-disable no-unused-expressions */
+  action === 'delete' && deleteTodo(todoId);
+  action === 'edit' && editTodo(todoId);
+});
+
+// to edit todo
+export function editTodo(todoId) {
+  const todoInput = document.getElementById('add-input');
+  todoInput.value = todos[todoId].value;
+  EditTodoId = todoId;
+}
```

Comment:
- [ ] Kindly fix this functionality, I wasn't able to edit the description of the selected todo item, instead it puts the text in the input field and clicking enter adds a new todo 👇 

![Animation](https://user-images.githubusercontent.com/50721479/206715671-c89fcea5-a1b9-4fb7-91e2-8d042a82c81c.gif)